dist/vendor.sh

```diff
@@ -26,8 +26,12 @@ cargo vendor > "${CARGO_CONFIG}"
 # some vendored files aren't world readable which has lead to issues in container builds when trying out the workflow
 find ./vendor -type f -exec chmod a+r {} \;
 
-git add --force "${FILES_TO_COMMIT[@]}"
-git diff --quiet --staged -- "${FILES_TO_COMMIT[@]}" || {
-  echo Files changed, committing...
-  git commit -m "Update vendored dependencies and rh-manifest.txt" -- "${FILES_TO_COMMIT[@]}"
-}
+# check the a variable as we don't want to commit new files on CI
+# it can also be set by developers which want to skip the git operations
+if [ "${OPENSHIFT_CI}" != "true" ]; then
```

Comment:
```suggestion
if [ "${OPENSHIFT_CI}" != "true" ]; then exit 0; fi
```

so that extra nesting would be avoided

Reply:
Nesting helps readability by making it obvious that the nested code is conditional. I regard it coincidental that this is code is at the end of the file. If the script is ever extended with more code after this condition your suggestion would require changing it just for the sake of being able to add more code.

Reply:
@vrutkovs you mean `if [ "${OPENSHIFT_CI}" = "true" ]; then exit 0; fi` right? you suggesting we should exit if `"${OPENSHIFT_CI}"` is set to true?

Reply:
> @vrutkovs you mean `if [ "${OPENSHIFT_CI}" = "true" ]; then exit 0; fi` right? you suggesting we should exit if `"${OPENSHIFT_CI}"` is set to true?

Note that your correcting the details of the suggestion is merely a distraction from the discussion I'm trying to have, which is about maintainability.

Reply:
Personally I will also prefer `if [ "${OPENSHIFT_CI}" = "true" ]; then exit 0; fi` as this is more easy to understand. But I this is not issue which should stop this PR from merging.

Reply:
Because it is more of a programming style.

Reply:
> Note that your correcting the details of the suggestion is merely a distraction from the discussion I'm trying to have, which is about maintainability.

I do not think maintainability is an issue with either styles if specifically think about this bash script.